openedx/core/djangoapps/credit/api/eligibility.py

```diff
@@ -179,12 +181,17 @@ def get_eligibilities_for_user(username):
     Returns: list
 
     """
+    eligibilities = CreditEligibility.get_user_eligibilities(username)
+
+    if course_key:
+        eligibilities = eligibilities.filter(course_key=course_key)
+
     return [
         {
             "course_key": eligibility.course.course_key,
             "deadline": eligibility.deadline,
         }
-        for eligibility in CreditEligibility.get_user_eligibilities(username)
+        for eligibility in eligibilities
     ]
```

Comment:
@wedaly I have updated this method for retrieving eligibility status and deadline of a user against particular course. We have to show these values on checkout page. What do you think that we should create separate API for retrieving this information or we should update our get_providers_detail() API, call this method and update response of API.